internal/cli/test.go

```diff
@@ -285,43 +252,68 @@ Specify the API you want this token for with --audience (API Identifer). Additio
 				cli,
 				tenant,
 				client,
-				"", // specifying a connection is only supported for the test login command
+				"", // Specifying a connection is only supported for the test login command.
 				inputs.Audience,
-				"", // We don't want to force a prompt for the test token command
+				"", // We don't want to force a prompt for the test token command.
 				inputs.Scopes,
-				"",
+				"", // Specifying a custom domain is only supported for the test login command.
 			)
 			if err != nil {
-				return fmt.Errorf("An unexpected error occurred when logging in to client %s: %w", inputs.ClientID, err)
-			}
-			if iostream.IsOutputTerminal() {
-				cli.renderer.GetToken(client, tokenResponse)
-			} else {
-				cli.renderer.Output(tokenResponse.AccessToken)
+				return fmt.Errorf("failed to log into the client %s: %w", inputs.ClientID, err)
 			}
+
+			cli.renderer.TestToken(client, tokenResponse)
+
 			return nil
 		},
 	}
 
 	cmd.SetUsageTemplate(resourceUsageTemplate())
 	cmd.Flags().BoolVar(&cli.force, "force", false, "Skip confirmation.")
 	cmd.Flags().BoolVar(&cli.json, "json", false, "Output in json format.")
-	testClientID.RegisterString(cmd, &inputs.ClientID, "")
 	testAudienceRequired.RegisterString(cmd, &inputs.Audience, "")
 	testScopes.RegisterStringSlice(cmd, &inputs.Scopes, nil)
 
 	return cmd
 }
 
-// cleanupTempApplication will delete the specified application if it is marked
-// as a temporary application. It will log success or failure to the user.
-func cleanupTempApplication(isTemp bool, cli *cli, id string) {
-	if isTemp {
-		if err := cli.api.Client.Delete(id); err != nil {
-			cli.renderer.Errorf("unable to remove the default test application", err.Error())
+func selectClientToUseForTestsAndValidateExistence(cli *cli, cmd *cobra.Command, args []string, inputs *testCmdInputs) (*management.Client, error) {
+	if len(args) == 0 {
+		if err := testClientID.Pick(cmd, &inputs.ClientID, cli.appPickerWithCreateOption); err != nil {
+			return nil, err
 		}
-		cli.renderer.Infof("Default test application removed")
+
+		if inputs.ClientID == newClientOption {
+			client := &management.Client{
+				Name:             auth0.String(cliLoginTestingClientName),
+				Description:      auth0.String(cliLoginTestingClientDescription),
+				Callbacks:        &[]string{cliLoginTestingCallbackURL},
+				InitiateLoginURI: auth0.String(cliLoginTestingInitiateLoginURI),
+			}
+
+			if err := cli.api.Client.Create(client); err != nil {
+				return nil, fmt.Errorf("failed to create a new client to use for testing the login: %w", err)
+			}
+
+			inputs.ClientID = client.GetClientID()
+
+			cli.renderer.Infof("New client created successfully.")
+			cli.renderer.Infof(
+				"If you wish to remove the created client after testing the login, run: 'auth0 apps delete %s'",
+				client.GetClientID(),
+			)
+			cli.renderer.Newline()
```

Comment:
At this point we could just return the client, because `client` will be hydrated by `Client.Create()`. So no need to do `Client.Read()` afterwards (in this case).

Reply:
Updated in [`03fac70` (#629)](https://github.com/auth0/auth0-cli/pull/629/commits/03fac70a8be5393dcaa4f7bbcb0643156266d357).